Enhance Positioning Options for Tooltips & Popovers

IdanCo opened the pull request:
### Summary:
Add more positioning options to the **tooltip** component, using _tether_'s native [attachment options](http://tether.io/#attachment). As requested in #17759 and #19921, and inspired by [semantic UI tooltips](https://semantic-ui.com/modules/popup.html#tooltip). No breaking changes, still fully support previous params. Docs updated accordingly.

### Before:
Tooltips supported only 4 positioning options: top | right | bottom | left

### After:
Tooltips now have an **optional** secondary alignment option, for example `top left` or `right bottom`, summing up in 12 positioning options.

IdanCo committed:
Expand positioning options for tooltips

Johann-S commented:
Hi @IdanCo, is it possible for you to post a JSBin with your modifications ?

IdanCo commented:
sure - http://jsbin.com/lijalowata/edit?html,output

I just copied the CSS and JS from the dist folder. Hope it's ok, couldn't think of a smarter way.

Johann-S commented:
IMO this PR is good to go 👍 

Any CSS feedbacks @mdo ? Or other feedbacks @bardiharborow  ?

NicoLB commented:
I think it then makes sense to also implement this on Popovers.

IdanCo commented:
@NicoLB , Sounds great!
@Johann-S , should it be in the same branch or as a different PR?

IdanCo committed:
add enhanced positioning to popover

IdanCo commented:
Added a new commit which implements the same enhancement to the popovers, namely add 8 more positions to the existing 4.

You can see it in action here - http://jsbin.com/voteco/edit?html,output

I've also updated the docs for popovers accordingly (similar to the update in tootltips) -
![image](https://cloud.githubusercontent.com/assets/10364499/25009246/bc0f2380-206e-11e7-9c28-be2620314421.png)

Johann-S commented:
Sorry but this PR is outdate with the merge of #22444 feel free to create a new one with Popper.js 😉
